### PR TITLE
[4.0]joomla.asset.json

### PR DESCRIPTION
**Pull request by brianteeman**
First draft at documenting the contents of the joomla.assets.json file to make it more intuitive for users to understand.

@fedik @wilsonge I hope I guessed these correctly - perhaps you can add the missing descriptions?

PARTIAL Pull Request for Issue #32408 - it does NOT close that discussion

### CODE REVIEW

File: templates/cassiopeia/joomla.asset.json [hunk 1]
```diff
@@ -37,6 +40,7 @@
     },
     {
       "name": "template.user",
+      "description": "The name of a file where a user can add their own css.",
```
Comment (wilsonge) [inline templates/cassiopeia/joomla.asset.json]:
```suggestion
      "description": "A file where a user can add their own css.",
```

File: templates/cassiopeia/joomla.asset.json [hunk 2]
```diff
@@ -46,6 +50,7 @@
     },
     {
       "name": "template.cassiopeia",
+      "description": "The name of the file containing the javascript for this template.",
```
Comment (wilsonge) [inline templates/cassiopeia/joomla.asset.json]:
```suggestion
      "description": "The file containing the javascript for this template.",
```

**Comment by wilsonge** [comment]
`template.active.language` as you can (kinda) see triggers the `\Joomla\CMS\WebAsset\AssetItem\LangActiveAssetItem` class which loads a CSS file for the language (should be obvious if you look at the class)

**Comment by Fedik** [comment]
the link additionally to  George comment:
https://github.com/joomla/joomla-cms/blob/421c3a15fd891c13559ea3f3ac7f5274aad6222f/libraries/src/WebAsset/AssetItem/LangActiveAssetItem.php#L16-L23

File: templates/cassiopeia/joomla.asset.json [hunk 3]
```diff
@@ -23,11 +25,13 @@
     },
     {
       "name": "template.offline",
+      "description": "The css file to be used when the site is offline and offline.php is being used.",
       "type": "style",
       "uri": "offline.css"
     },
     {
       "name": "template.active.language",
+      "description": "A dummy asset to allow language specific css to use it as a dependency to the active template",
```
Comment (Fedik) [inline templates/cassiopeia/joomla.asset.json]: not a dummy, it very "active" one ;)

Comment (Fedik) [reply templates/cassiopeia/joomla.asset.json]: just:
`An asset to allow language specific css ....`

**Comment by Fedik** [comment]
Please copy it also to an administrator template. To be consistent.

File: templates/cassiopeia/joomla.asset.json (hunk 3 above)
Comment (brianteeman) [reply templates/cassiopeia/joomla.asset.json]: I wrote dummy as it looked the same as template.active

Comment (Fedik) [reply templates/cassiopeia/joomla.asset.json]: no no, it totally different, 
`template.active` is empty itself, and provide only "dependency"
but
`template.active.language` is doing a job, it looking for language/[lang-CODE]/[lang-CODE].css file

**Comment by ceford** [comment]
I have an interest in this topic. I have created a custom site template based on Cassiopeia but using Bootstrap 5 when Joomla was on BS4. I read the docs referred to and would have found the json descriptions helpful. Please carry on. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32411">issues.joomla.org/tracker/joomla-cms/32411</a>.</sub>

**Comment by dgrammatiko** [comment]
@ceford fwiw the js of BS5 **is** compatible with BS4 as long as you use the new `data-bs-*` convention. When I did the conversion I also used the BS4 css it was @wilsonge's PR that brought the BS5 css to Joomla. So what you're doing is fine and maybe should be some kind of unofficially supported template as BS4 still supports IE11, but that's not my call

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on acff04231dd7962f415cf8a1e7113aa3fc013e30<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32411">issues.joomla.org/tracker/joomla-cms/32411</a>.</sub>

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on acff04231dd7962f415cf8a1e7113aa3fc013e30<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32411">issues.joomla.org/tracker/joomla-cms/32411</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32411">issues.joomla.org/tracker/joomla-cms/32411</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
thank you

**Pull request merged**